src/iop/ashift.c

```diff
@@ -373,6 +373,7 @@ typedef struct dt_iop_ashift_cropfit_params_t
 
 typedef struct dt_iop_ashift_gui_data_t
 {
+  dt_iop_gui_data_t common; // contains required fields such as .lock
```

Comment:
won't it work to put those fields into dt_iop_module_t directly, a common set of gui_data? This was there will be no need for each module to define this common field and all fields inside will be directly accessed via imageop API. I think this will simplify this business quite a bit.

Reply:
Sure, that would work too.  And now that all accesses to the `lock` field have been centralized, it's trivial to switch over.  I'll rename `lock` to `gui_lock` though, so that its purpose is clear.